frontend2/src/views/Scrimmaging.tsx

```diff
@@ -0,0 +1,542 @@
+import React, { useEffect, useMemo, useState } from "react";
+import type {
+  PaginatedScrimmageRequestList,
+  PaginatedMatchList,
+  PaginatedTeamPublicList,
+  TeamPublic,
+  ScrimmageRequest,
+} from "../utils/types/models";
+import { NavLink, useSearchParams } from "react-router-dom";
+import { searchTeams } from "../utils/api/team";
+import {
+  getScrimmagesByTeam,
+  getTournamentMatches,
+  getUserScrimmagesInbox,
+  getUserScrimmagesOutbox,
+} from "../utils/api/compete";
+import Input from "../components/elements/Input";
+import Button from "../components/elements/Button";
+import BattlecodeTable from "../components/BattlecodeTable";
+import BattlecodeTableBottomElement from "../components/BattlecodeTableBottomElement";
+import MatchScore from "../components/compete/MatchScore";
+import RatingDelta from "../components/compete/RatingDelta";
+import MatchStatus from "../components/compete/MatchStatus";
+import { useEpisodeId } from "../contexts/EpisodeContext";
+import { useCurrentTeam } from "../contexts/CurrentTeamContext";
+import { dateTime } from "../components/compete/DateTime";
+import Collapse from "../components/elements/Collapse";
+
+function trimString(str: string, maxLength: number): string {
+  if (str.length > maxLength) {
+    return str.slice(0, maxLength - 1) + "...";
+  }
+  return str;
+}
+
+const Scrimmaging: React.FC = () => {
+  const episodeId = useEpisodeId().episodeId;
+  const currentTeam = useCurrentTeam().team;
+
+  const [searchParams, setSearchParams] = useSearchParams();
+  const getParamEntries = (prev: URLSearchParams): Record<string, string> => {
+    return Object.fromEntries(prev.entries());
+  };
+
+  interface QueryParams {
+    inboxPage: number;
+    outboxPage: number;
+    teamsPage: number;
+    scrimsPage: number;
+    tourneyPage: number;
+    search: string;
+  }
```

Comment:
nit: move this interface definition outside the component function b/c it doesn't depend on component state